### ISSUE TITLE
Feature request: Agents to controller connection

### ISSUE THREAD

**Issue by mocenas** [Hyperfoil/Hyperfoil#56]
Current way of connection controller to agents via SSH seems a bit clumsy. Especially defining the agents in benchmark.

We propose to connect agents to controller, via their own protocol (or at least using their own port). 
From user's point of view, they will do:
1. Start controller - define port, maybe password, where the controller should listen.
2. Start agents - on each agent start a hyperfoil-agent script/binary/whatever and just configure a controller's address, port etc. on it.
Agents will then connect to the controller and form a cluster. 
To run a tests, client will just connect to controller a send a benchmark to it. Benchmark should contain only number of agents (maybe agents can have tags?) to run on.

Is it feasible to configure a test cluster this way?

**Comment by rvansa** [comment]
@mocenas I am perfectly fine with running a small piece of code (part of HF codebase) that will open some port and on custom request spawn the agent as a separate (child) process, if you find SSH too clumsy. Reusing the agent process sounds a bit risky; it's easier to let OS do all the cleanup (possibly leaked connections).

However I am quite opposed to *not keep* the agent list in the benchmark. We aim at maximum reproducibility and having the agents written down in the definition is more bullet proof; that's the reason why we went for that option. No 2 sets of 4 agent nodes will perform exactly same.

If you're working in an environment when the agent list changes, I suggest using external template engine to fill it up and reupload the benchmark definition: see e.g. https://github.com/RedHatPerf/scale-test/blob/master/benchmarks/test.yaml.j2 We've discussed including some templating options directly into HF, but didn't come up with a definite answer.
